libc/test/src/string/strerror_test.cpp

```diff
@@ -6,153 +6,16 @@
-      "Bad font file format",
-      "Device not a stream",
-      "No data available",
-      "Timer expired",
-      "Out of streams resources",
-      "Machine is not on the network",
-      "Package not installed",
-      "Object is remote",
-      "Link has been severed",
-      "Advertise error",
-      "Srmount error",
-      "Communication error on send",
-      "Protocol error",
-      "Multihop attempted",
-      "RFS specific error",
-      "Bad message",
-      "Value too large for defined data type",
-      "Name not unique on network",
-      "File descriptor in bad state",
-      "Remote address changed",
-      "Can not access a needed shared library",
-      "Accessing a corrupted shared library",
-      ".lib section in a.out corrupted",
-      "Attempting to link in too many shared libraries",
-      "Cannot exec a shared library directly",
-      "Unknown error 84", // Unknown
-      "Interrupted system call should be restarted",
-      "Streams pipe error",
-      "Too many users",
-      "Socket operation on non-socket",
-      "Destination address required",
-      "Message too long",
-      "Protocol wrong type for socket",
-      "Protocol not available",
-      "Protocol not supported",
-      "Socket type not supported",
-      "Operation not supported",
-      "Protocol family not supported",
-      "Address family not supported by protocol",
-      "Address already in use",
-      "Cannot assign requested address",
-      "Network is down",
-      "Network is unreachable",
-      "Network dropped connection on reset",
-      "Software caused connection abort",
-      "Connection reset by peer",
-      "No buffer space available",
-      "Transport endpoint is already connected",
-      "Transport endpoint is not connected",
-      "Cannot send after transport endpoint shutdown",
-      "Too many references: cannot splice",
-      "Connection timed out",
-      "Connection refused",
-      "Host is down",
-      "No route to host",
-      "Operation already in progress",
-      "Operation now in progress",
-      "Stale file handle",
-      "Structure needs cleaning",
-      "Not a XENIX named type file",
-      "No XENIX semaphores available",
-      "Is a named type file",
-      "Remote I/O error",
-      "Disk quota exceeded",
-      "No medium found",
-      "Wrong medium type",
-      "Operation canceled",
-      "Required key not available",
-      "Key has expired",
-      "Key has been revoked",
-      "Key was rejected by service",
-      "Owner died",
-      "State not recoverable",
-      "Operation not possible due to RF-kill",
-      "Memory page has hardware error",
-  };
-
-  for (size_t i = 0; i < (sizeof(message_array) / sizeof(char *)); ++i) {
-    EXPECT_STREQ(LIBC_NAMESPACE::strerror(static_cast<int>(i)),
-                 message_array[i]);
-  }
+  for (auto [i, msg] : LIBC_NAMESPACE::PLATFORM_ERRORS)
+    EXPECT_STREQ(LIBC_NAMESPACE::strerror(static_cast<int>(i)), msg.begin());
```

Comment:
I'm somewhat hesitant to do it this way, since it's effectively comparing the values against themselves, but I can't think of a better solution.